compiler/rustc_session/src/options.rs

```diff
@@ -158,6 +158,7 @@ top_level_options!(
         /// can influence whether overflow checks are done or not.
         debug_assertions: bool [TRACKED],
         debuginfo: DebugInfo [TRACKED],
+        debug_name_table_kind: DebugNameTableKind [TRACKED],
```

Comment:
If this function is introduced, then it should start as an unstable `-Z debug-name-table-kind` flag instead of `--debug-name-table-kind`.

Reply:
Also, in this PR the option is never parsed from the command line to the `Options` structure.